### PR TITLE
feat: add kakarot madara swap starksheet

### LINKED ISSUE #7: stack: add kakarot+madara swap+starkshet to starknet stack
Add it to the compose file

ref: #4

### PR DESCRIPTION
**Pull request by ftupas**
## PR Overview
Hello, this pr does the following:

- [x] add Kakacet front-end to fund corresponding Starknet address of an Ethereum address, it uses the starkcet backend
- [x] connect Kakarot RPC to the Madara instance
- [x] use committed images from the official repos

Fix #7 

### Dependencies
- https://github.com/kkrt-labs/kakarot-rpc/pull/314
- https://github.com/kkrt-labs/kakacet/pull/3

### CODE REVIEW

File: starknet-stack/docker-compose.yml [hunk 1]
```diff
@@ -36,22 +36,71 @@ services:
...
+      - starkcet-back
+    networks:
+      - internal
+
+  kakarot:
+    <<: *deploy
+    build:
+      dockerfile: misc-apps/kakarot/Dockerfile
+    volumes:
+      - ./misc-apps/kakarot/deployments:/app/kakarot/deployments
```
Comment (ftupas) [inline starknet-stack/docker-compose.yml:83]: Here I split the applications to their own containers to deploy in parallel

File: starknet-stack/.env [hunk 2]
```diff
@@ -0,0 +1,2 @@
+RPC_URL=http://madara:9944
+STARKNET_NETWORK=sharingan
```
Comment (ftupas) [inline starknet-stack/.env:2]: Use `.env` to sync the environment variables

File: starknet-stack/.gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 
 madara_explorer/
 substrate-telemetry/
+deployments/
```
Comment (ftupas) [inline starknet-stack/.gitignore:4]: Class hashes and deployed contract addresses are found here

File: starknet-stack/misc-apps/deploy.sh [hunk 4]
```diff
@@ -1,13 +1,10 @@
...
+export "${SN}_ACCOUNT_ADDRESS"=0x3
+export "${SN}_PRIVATE_KEY"=0x00c1cf1490de1352865301bb8705143f3ef938f97fdf892f1090dcb5ac7bcd1d
 
-cd /app/kakarot && make deploy &
-cd /app/starksheet-monorepo/packages/starksheet-cairo && poetry run python deploy/starksheet.py
-cd /app/madara-swap/packages/contracts && make deploy
-
-# Wait for both commands to finish
-wait
+exec "$@"
```
Comment (ftupas) [inline starknet-stack/misc-apps/deploy.sh:10]: Script used for parameterize deployment

File: starknet-stack/misc-apps/kakarot/run.sh [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+export KAKAROT_ADDRESS=$(jq -r '.kakarot.address' ./deployments/${STARKNET_NETWORK}/deployments.json)
+export PROXY_ACCOUNT_CLASS_HASH=$(jq -r '.proxy' ./deployments/${STARKNET_NETWORK}/declarations.json)
+exec "$@"
```
Comment (ftupas) [inline starknet-stack/misc-apps/kakarot/run.sh:5]: This is needed for the RPC to initialize

**Comment by ClementWalter** [comment]
Ok so I've created a bunch of Docker images in their corresponding repo. All the images put info in `/app/{app_name}/...` so using a single `volume` should be easy to get info from app into another

- [madara-swap deployer image]https://github.com/keep-starknet-strange/madara-swap/pkgs/container/madara-swap%2Fcontracts)

```
docker pull ghcr.io/keep-starknet-strange/madara-swap/contracts:latest
```

- [starksheet deployer image](https://github.com/the-candy-shop/starksheet-monorepo/pkgs/container/starksheet-monorepo%2Fstarksheet-cairo)

```
docker pull ghcr.io/the-candy-shop/starksheet-monorepo/starksheet-cairo:latest
```

- [starksheet webapp image](https://github.com/the-candy-shop/starksheet-monorepo/pkgs/container/starksheet-monorepo%2Fstarksheet-webapp)

```
docker pull ghcr.io/the-candy-shop/starksheet-monorepo/starksheet-webapp:latest
```

- see [starksheet compose file](https://github.com/the-candy-shop/starksheet-monorepo/blob/staging/docker-compose.yml) for an example of how to run the webapp after contract deployment

**Comment by ClementWalter** [comment]
see with @ftupas; let's close this one in favor of https://github.com/keep-starknet-strange/madara-infra/pull/16

**Pull request closed without merging**